app/models/template.rb

```diff
@@ -53,24 +53,25 @@ def default
       unarchived.where(is_default: true, published: true).order(:version).last
     end
   end
-
-  # Creates a copy of the current template
-  # raises ActiveRecord::RecordInvalid when save option is true and validations fails
-  def deep_copy(**options)
-    copy = self.dup
-    copy.version = options.fetch(:version, self.version)
-    copy.published = options.fetch(:published, self.published)
-    copy.save! if options.fetch(:save, false)
-    self.phases.each{ |phase| copy.phases << phase.deep_copy(options) }
-    return copy
-  end
-
+  
   # Returns whether or not this is the latest version of the current template's family
   def is_latest?
     return (self.id == Template.latest_version(self.family_id).pluck(:id).first)
   end
 
-  # Generates a new copy of self
+  # Returns a new unpublished copy of self with a new family_id, version = zero for the specified org
+  def generate_copy(org)
+    template = deep_copy(modifiable: true, version: 0, published: false, save: true)
+    template.update!({
```

Comment:
I'd rather prefer to to send all the attributes at once when we create the template so that we don't end up with temporary template records that are not yet valid. I'm going to extend the deep_copy for template